### ISSUE TITLE
hl7800  configure_TCP_socket for tls connection

### ISSUE THREAD

**Issue by zafersn** [LairdCP/BLE_Gateway_Firmware#1]
Hi
I was looking at this [example ](https://github.com/LairdCP/BLE_Gateway_Firmware/blob/gateway/app/common/src/lcz_mqtt.c) and little bit get confused here: 

when you try to have aws tls connection over the lte (hl78). it seems basically you called configure_TCP_socket func. in the hl7800 but  [here ](https://github.com/LairdCP/sdk-zephyr/blob/3041adb35158975b4525aeadef6d026f866d6685/drivers/modem/hl7800.c#L5662)

doesn't this config allow you to have only a non secure client connection? 

snprintk(cmd_cfg, sizeof(cmd_cfg), "AT+KTCPCFG=%d,%d,\"%s\",%u,,,,%d,,%d", 1, 0,
		 hl7800_sprint_ip_addr(&sock->dst), dst_port, af, restore_on_boot);

basically you are setting AT+KTCPCFG=1,0 and 2. parameter is the mode that selected a non secure client.

Am I missing something, can you please shed some light on this point?

**Comment by rerickson1** [comment]
All TLS is handled by the Zephyr stack, not the HL7800 sockets.
